src/line.rs

```diff
@@ -1062,6 +1123,72 @@ impl<'input, Endian> LineNumberProgramHeader<'input, Endian>
         StateMachine::new(self)
     }
 
+    /// Execute the line number program, completing the LineNumberProgramHeader
+    /// and producing an array of sequences within the line number program that
+    /// can later be used with `resume_from`.
+    ///
+    /// ```
+    /// # fn foo() {
+    /// use gimli::{LineNumberProgramHeader, NativeEndian};
+    ///
+    /// fn get_line_number_program_header<'a>() -> LineNumberProgramHeader<'a, NativeEndian> {
+    ///     // Get a line number program header from some offset in a
+    ///     // `.debug_line` section...
+    /// #   unimplemented!()
+    /// }
+    ///
+    /// let header = get_line_number_program_header();
+    /// let (complete_header, sequences) = header.sequences().unwrap();
+    /// println!("There are {} sequences in this line number program", sequences.len());
+    /// # }
+    /// ```
+    pub fn sequences(self)
+                     -> parser::Result<(LineNumberProgramHeader<'input, Endian>,
+                                        Vec<LineNumberSequence<'input, Endian>>)> {
```

Comment:
This is the only comment I haven't addressed yet.  Having the caller iterate through the sequences runs the risk of producing `LineNumberSequence`s that can't actually be used yet if the `StateMachine` has not run to completion.

An outparam `Vec` doesn't have that problem, but we'd have to be careful to clear the contents of the `Vec` if we return early due to a parser error condition.